### PR TITLE
APPS-1157 symlink download improvements

### PR DESCRIPTION
**Pull request by mhrvol**
Based on https://github.com/dnanexus/dx-toolkit/pull/709

### CODE REVIEW

File: src/python/dxpy/bindings/dxfile_functions.py [hunk 1]
```diff
@@ -175,65 +178,53 @@ def _verify(filename, md5digest):
 
 # [dxid] is a symbolic link. Create a preauthenticated URL,
 # and download it
-def _download_symbolic_link(dxid, md5digest, project, dest_filename):
+def _download_symbolic_link(dxid, md5digest, project, dest_filename, symlink_max_tries=15):
+    print(symlink_max_tries)
+    if symlink_max_tries < 1:
```
Comment (mhrvol) [inline src/python/dxpy/bindings/dxfile_functions.py:183]: symlink_max_tries is currently checked in the download parser, so there is no way this code will ever be triggered. I added this code just to make sure that if somebody ever uses this, we will never get infinite loop (which would be case for symlink_max_tries == 0 if the file is for some reason 'undownloadable')

File: src/python/dxpy/bindings/dxfile_functions.py [hunk 2]
```diff
@@ -97,7 +97,7 @@ def new_dxfile(mode=None, write_buffer_size=dxfile.DEFAULT_BUFFER_SIZE, expected
 
 
 def download_dxfile(dxid, filename, chunksize=dxfile.DEFAULT_BUFFER_SIZE, append=False, show_progress=False,
-                    project=None, describe_output=None, **kwargs):
+                    project=None, describe_output=None, symlink_max_tries=False, **kwargs):
```
Comment (commandlinegirl) [inline src/python/dxpy/bindings/dxfile_functions.py:100]: The default should not be boolean.

File: src/python/dxpy/scripts/dx.py [hunk 3]
```diff
@@ -4566,6 +4571,9 @@ def register_parser(parser, subparsers_action=None, categories=('other', ), add_
                              action='store_false', default=sys.stderr.isatty())
 parser_download.add_argument('--lightweight', help='Skip some validation steps to make fewer API calls',
                              action='store_true')
+parser_download.add_argument('--symlink-max-tries', help='Set maximum tries for downloading symlinked files using aria2c',
```
Comment (commandlinegirl) [inline src/python/dxpy/scripts/dx.py:4574]:
```suggestion
parser_download.add_argument('--symlink-max-tries', help='Set maximum number of tries for downloading symlinked files using aria2c',
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -6,6 +6,10 @@ Categories for each release: Added, Changed, Deprecated, Removed, Fixed, Securit
 
 ## Unreleased
 
+## Fixed
+
+* Improvements to symlink downloading reliability by solely using `aria2c` and enhancing options around its use
```
Comment (commandlinegirl) [inline CHANGELOG.md:11]:
```suggestion
* Improvements to symlink downloading reliability by solely using `aria2c` and enhancing options around its use (removes `wget` option for downloading symlinked files, adds the ability to set max tries for aria2c, adds `-c` flag for continuing downloads, removes the `--check-certificate=false` option).
```

**Review by commandlinegirl — CHANGES_REQUESTED** [review]
Minor requests. Thanks!

File: src/python/dxpy/bindings/dxfile_functions.py (hunk 2 above)
Comment (mhrvol) [reply src/python/dxpy/bindings/dxfile_functions.py]: Thanks! fixed

**Review by commandlinegirl — APPROVED** [review]
LGTM, thanks!

**Pull request merged**